### PR TITLE
Introduce AL2023 to CI tests

### PR DESCRIPTION
**Pull request by Kausik-A**
**Description:** 

Introduce AL2023 to CI tests. These changes were already bought to the dev branch and tested on the ADOt CI.

https://github.com/aws-observability/aws-otel-test-framework/pull/1413

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: terraform/ec2/main.tf [hunk 1]
```diff
@@ -405,9 +405,11 @@ resource "null_resource" "setup_sample_app_and_mock_server" {
   provisioner "remote-exec" {
     inline = [
       "sudo yum update -y",
-      "sudo amazon-linux-extras install docker -y",
+      "sudo yum install -y libxcrypt-compat",
+      "sudo yum install -y docker",
       "sudo service docker start",
       "sudo usermod -a -G docker ec2-user",
+      "sudo ln -s /usr/lib/libcrypt.so /usr/lib/libcrypt.so.1",
```
Comment (bryan-aguilar) [inline terraform/ec2/main.tf]: Do we need this anymore since we are not updating sample app to al2023?

File: terraform/ec2/variables.tf [hunk 2]
```diff
@@ -21,7 +21,7 @@ variable "package_s3_bucket" {
 }
 
 variable "testing_ami" {
-  default = "amazonlinux2"
+  default = "amazonlinux3"
 }
```
Comment (bryan-aguilar) [inline terraform/ec2/variables.tf]: What is this default changing? For the sample app or the collector? Either way, I don't think we should change the default.

File: terraform/ec2/main.tf (hunk 1 above)
Comment (Kausik-A) [reply terraform/ec2/main.tf]: We dont need them but it does not change any behavior on Al2022. Skipping from using `amazon-linux-extras` will make sure the future engineer not face the same issue when they try to upgrade the sample app instances to AL2023

Comment (bryan-aguilar) [reply terraform/ec2/main.tf]: This didn't even work though right? There were additional issues on top? I would prefer to just remove this if it's not necessary. You can leave it as a comment if you'd like.

File: terraform/ec2/variables.tf (hunk 2 above)
Comment (Kausik-A) [reply terraform/ec2/variables.tf]: The collector. But yeah, I agree. I changed default back to AL2022

**Comment by bryan-aguilar** [comment]
This is missing required changes to batch test generator. https://github.com/aws-observability/aws-otel-test-framework/blob/c869895c95b5d52b9089bff1596ca49d1bd9bda5/tools/batchTestGenerator/internal/testCaseBuilder.go#L8C2-L8C2

File: terraform/ec2/main.tf (hunk 1 above)
Comment (Kausik-A) [reply terraform/ec2/main.tf]: They worked but there were additional issues relating to docker login. Okay, I have reverted the changes.

File: terraform/ec2/main.tf [hunk 3]
```diff
@@ -405,7 +405,7 @@ resource "null_resource" "setup_sample_app_and_mock_server" {
   provisioner "remote-exec" {
     inline = [
       "sudo yum update -y",
-      "sudo amazon-linux-extras install docker -y",
+      "sudo amazon-linux-extra install docker -y",
```
Comment (bryan-aguilar) [inline terraform/ec2/main.tf]: So the original command was wrong? Or are they different?

Comment (Kausik-A) [reply terraform/ec2/main.tf]: oops, fixed it

**Review by bryan-aguilar — APPROVED** [review]

**Review by vasireddy99 — APPROVED** [review]

**Pull request merged**